crates/iceberg/src/spec/values.rs

```diff
@@ -1309,4 +1928,256 @@ mod tests {
+                    Some(Literal::Primitive(PrimitiveLiteral::Int(2))),
+                ),
+                (
+                    Literal::Primitive(PrimitiveLiteral::String("c".to_string())),
+                    None,
+                ),
+            ])),
+            &Type::Map(MapType {
+                key_field: NestedField::map_key_element(0, Type::Primitive(PrimitiveType::String))
+                    .into(),
+                value_field: NestedField::map_value_element(
+                    1,
+                    Type::Primitive(PrimitiveType::Int),
+                    false,
+                )
+                .into(),
+            }),
+        );
+
+        check_convert_with_avro(
+            Literal::Map(BTreeMap::from([
+                (
+                    Literal::Primitive(PrimitiveLiteral::String("a".to_string())),
+                    Some(Literal::Primitive(PrimitiveLiteral::Int(1))),
+                ),
+                (
+                    Literal::Primitive(PrimitiveLiteral::String("b".to_string())),
+                    Some(Literal::Primitive(PrimitiveLiteral::Int(2))),
+                ),
+                (
+                    Literal::Primitive(PrimitiveLiteral::String("c".to_string())),
+                    Some(Literal::Primitive(PrimitiveLiteral::Int(3))),
+                ),
+            ])),
+            &Type::Map(MapType {
+                key_field: NestedField::map_key_element(0, Type::Primitive(PrimitiveType::String))
+                    .into(),
+                value_field: NestedField::map_value_element(
+                    1,
+                    Type::Primitive(PrimitiveType::Int),
+                    true,
+                )
+                .into(),
+            }),
+        );
+    }
+
+    #[test]
+    fn avro_convert_test_record() {
+        check_convert_with_avro(
+            Literal::Struct(Struct::from_iter(vec![
+                (
+                    1,
+                    Some(Literal::Primitive(PrimitiveLiteral::Int(1))),
+                    "id".to_string(),
+                ),
+                (
+                    2,
+                    Some(Literal::Primitive(PrimitiveLiteral::String(
+                        "bar".to_string(),
+                    ))),
+                    "name".to_string(),
+                ),
+                (3, None, "address".to_string()),
+            ])),
+            &Type::Struct(StructType::new(vec![
+                NestedField::required(1, "id", Type::Primitive(PrimitiveType::Int)).into(),
+                NestedField::optional(2, "name", Type::Primitive(PrimitiveType::String)).into(),
+                NestedField::optional(3, "address", Type::Primitive(PrimitiveType::String)).into(),
+            ])),
+        );
+    }
+
+    // # TODO
+    // rust avro don't support deserialize any bytes representation now:
+    // - binary
+    // - fixed
+    // - decimal
+    // - uuid
```

Comment:
I think we can still test them? They are encoded in bytes when converting from literal to raw literal?

Reply:
Did you means we can test serialize?